src/lightning/fabric/fabric.py

```diff
@@ -756,10 +753,9 @@ def seed_everything(seed: Optional[int] = None, workers: Optional[bool] = None)
 
     def _run_impl(self, run_method: Callable, *args: Any, **kwargs: Any) -> Any:
```

Comment:
With run_impl I wanted to express that the original run() method becomes replaced by this `setattr(self, "run", partial(self._run_impl, self.run))`. Even if the code is the same, it is conceptually different. In `.launch()`, we want to launch the function. I think it is confusing to mix run_method and function. I am in favor of keeping it separate for readability (but the walrus simplification is nice).

Reply:
But that setattr happens on a different method: `_prepare_run_method`

This is just an utility function. Would renaming it make it clearer? Either way I'm fine closing this PR if you prefer to keep the duplication.

Reply:
I think passing the function in launch() into a wrapper that is meant to go around the run method is just strange, since launch(function) has nothing to do with the run method. 

If you still want to do it, the only way I can think of this could make sense is to rename it to something else, wrap_and_launch, and change the local variable name from run_method to function